int warning_type
int direction
int severity